### ISSUE TITLE
Service dependency

### ISSUE THREAD

**Issue by udragon** [arachnys/cabot#410]
Hey guys, do cabot has service dependency feature? In most cases when one service is broken other checks will also fail.

**Comment by dbuxton** [comment]
Well, the point of services in general is that they bundle up lots of individual checks which may fail independently or together.

One problem with service dependencies is that few dependency outages result in a 100% of the consumer service (although some do).

What feature would you want to see?

**Comment by udragon** [comment]
I'll give an example - Let's say the network service is down. Well, in this case I would get dozens of notifications saying all services failed, although I would like to get only one notification - "Network is down.".

**Comment by dbuxton** [comment]
So you would want upstream services to squash downstream services' alerts? Your hypothetical example is reasonable in a scenario like network (where you have different layers of protocols that are strict supersets of underlying services - e.g. no point worrying about TCP/IP if someone puts a backhoe through your cable) but I'm not sure it maps well onto the vast majority of real-world scenarios that most people are using Cabot to track.

Happy to be corrected though!

**Comment by udragon** [comment]
There are quite a few components which will affect other components on failure. Another examples will be databases and storage. 
I would like to point out that most monitoring projects do have this feature, e.g icinga, nagios, sensu and shinken.

**Comment by dbuxton** [comment]
Certainly, if you can propose a way in which it would work, I'd be happy to look at it. To some extent this is the whole point of having both **Services** and **Checks** - which was a less sophisticated way than formal dependency trees, but easier to work with in practice.
